.github/workflows/haskell.yml

```diff
@@ -129,26 +129,45 @@ jobs:
     - name: Build all
       run: cabal build all --enable-tests
 
-    # This step is temporary. We'll soon remove it and check instead
-    # that golden files have not been modified when creating them below.
-    - name: Test all
-      run: cabal test all --enable-tests --test-show-details=direct -j1
-
     - # Delete golden files and regenerate them in the next step to ensure
       # files that tests designate as golden files are actually golden files
       # and can be generated.
       name: Delete golden files
       run: |
         rm -rf cardano-cli/test/cardano-cli-golden/files/golden
 
-    - name: Run tests regenerate golden files
+    - name: Run tests, generating golden files on the fly
       env:
         TMPDIR: ${{ runner.temp }}
         TMP: ${{ runner.temp }}
         KEEP_WORKSPACE: 1
         CREATE_GOLDEN_FILES: 1
       run: cabal test all --enable-tests --test-show-details=direct -j1
 
+    # We want this check to run first because $(git ls-files -m) (see beow) returns both
+    # modified files *and* deleted files. So we want to fail on deleted files first.
+    # This makes sure we only report modified files (and not deleted ones) in the next step,
+    # which is more intuitive.
+    - name: Check golden files are all being used
+      run: |
+        NB_UNUSED_GOLDEN_FILES=$(git ls-files -d | wc -l)
+        if [[ "$NB_UNUSED_GOLDEN_FILES" != "0" ]]; then
+          echo "⚠️ The following golden files are not used anymore:"
+          git ls-files -d
+          echo "Please delete them."
+          exit 1
```

Comment:
Seems this is a problem for windows: https://github.com/IntersectMBO/cardano-cli/blob/main/cardano-cli/test/cardano-cli-golden/Test/Golden/Help.hs#L54

Reply:
Yep, I simply disabled the new checks on Windows: https://github.com/IntersectMBO/cardano-cli/commit/ad0b99acdc0afd25fbaab57c148aea5a92f665a0